src/app/_components/followButton/FollowButton.tsx

```diff
@@ -1,33 +1,68 @@
+import { useClerk } from "@clerk/nextjs";
 import { Button } from "@mui/material";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { type FollowButtonProps } from "~/app/(with-sidebar)/myfeed/types/types";
+import { handleError } from "~/app/_utils/handleError";
+import { trpc } from "~/server/client";
 
 const FollowButton: React.FC<FollowButtonProps> = ({
   authorProfileLink,
-  yPadding = "3px",
-  xPadding = "4px",
+  style,
+  followState = "Follow",
 }) => {
-  const followWriter = (authorProfileLink: string) => {
-    console.log(`followed writer ${authorProfileLink} `);
+  const { user } = useClerk();
+
+  const [title, setTitle] = useState(followState);
+  const userMutation = trpc.user;
+  const followUser = userMutation.followUser.useMutation();
+
+  const followWriter = async () => {
+    const userEmail = user?.primaryEmailAddress?.emailAddress;
+
+    if (userEmail) {
+      try {
+        const res = await followUser.mutateAsync({
+          currentUserEmail: userEmail,
+          followerId: authorProfileLink,
+        });
+
+        if (res?.message === "Unfollowed") {
+          setTitle("Follow");
+        } else {
+          setTitle("Following");
+        }
+      } catch (error) {
+        handleError(error);
+        throw error;
+      }
+    }
   };
 
+  useEffect(() => {
```

Comment:
Why do we need this useEffect? Can't we handle this in handleFollow callback function

Reply:
done